pyslim/methods.py

```diff
@@ -608,6 +608,30 @@ def annotate_tables(tables, model_type, tick, cycle=None, stage="early", referen
     if reference_sequence is not None:
         tables.reference_sequence.data = reference_sequence
 
+def next_slim_mutation_id(ts):
+    '''
+    Returns the next SLiM mutation ID for this tree sequence. This is useful 
+    because if you want to add more mutations to your SLiM tree sequence using 
+    :func:`msprime.sim_mutations`, you may need to specify the parameter 
+    `next_id` in your :class:`msprime.SLiMMutationModel` to be larger than any 
+    existing mutation IDs. Setting `next_id` equal to the output of this 
+    function will allow the mutated tree sequence to be read in by SLiM.
+    To do this, recall that the "derived state" of SLiM's mutations are
+    comma-separated strings of mutation IDs; this function just parses all derived
+    states and returns one larger than the largest integer found. It will return an error
+    if it encounters derived states that are not comma-separated strings of integers.
+    '''
+    max_id = 0
+    for mut in ts.mutations():
+        for d in mut.derived_state.split(","):
+            try:
+                max_id = max(max_id, int(d or 0)) # the or zero will allow empty strings to be converted
+            except ValueError:
+                raise ValueError("The derived states of mutations in the tree "
+                                 "sequence need to be coercible to int. This "
+                                 "is not a valid SLiM tree sequence.")
```

Comment:
Gee, that's clever. Hm, but the only time we'd get an empty string is if the derived state is `""` (so e.g., `"9,"` is not valid). So, how about this, which will correctly not work with `"9,"` and is a bit more clear that it's right, also?
```suggestion
        ds = mut.derived_state
        if len(ds) > 0:
            for d in mut.derived_state.split(","):
                try:
                    max_id = max(max_id, int(d))
                except ValueError:
                    raise ValueError("The derived states of mutations in the tree "
                                     "sequence need to be coercible to int. This "
                                     "is not a valid SLiM tree sequence.")
```